src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp

```diff
@@ -602,6 +639,68 @@ class TDatabaseFeedManager final : public Observer<nlohmann::json&>
     std::unique_ptr<LRUCache<std::string, std::string>> m_translationsCache;
     std::unique_ptr<TRouterSubscriber> m_contentUpdateSubscription;
     const std::atomic<bool>& m_shouldStop;
+    std::unordered_map<std::string, std::string> m_prefixVendorMap;
+    std::unordered_map<std::string, std::string> m_containsVendorMap;
+    std::unordered_map<std::string, std::string> m_formatVendorMap;
+    std::atomic<bool> m_isVendorMapValid {false};
+
+    /**
+     * @brief Reads the vendor maps from the database and loads the data into memory.
+     *
+     * @throws std::runtime_error if the vendor map isn't available.
+     * @note This methods locks the mutex.
+     */
+    void reloadVendorMaps()
+    {
+        std::scoped_lock<std::shared_mutex> lock(m_mutex);
+        m_isVendorMapValid = false;
+        if (!m_feedDatabase->columnExists(VENDOR_MAP_COLUMN))
```

Comment:
I think we are making things complex because of the format that comes with intelligence data.

I want us here to simply read a json, keep the nlohmann json in memory and in each getter we loop through the json depending on the type.

Reply:
```json
{
  "prefix": [
    {"Canonical": "canonical"},
    {"Ubuntu": "canonical"},
    {"Debian": "debian"},
    {"Red Hat, Inc.": "redhat"},
    {"CentOS": "redhat"},
    {"Amazon Linux": "alas"},
    {"Amazon.com": "alas"},
    {"Amazon AWS": "alas"},
    {"Arch Linux": "arch"},
    {"suse": "suse"},
    {"AlmaLinux": "almalinux"},
    {"CloudLinux": "almalinux"}
  ],
  "contains": [
    {"@ubuntu.com": "canonical"}
  ],
  "format": [
    {"pypi": "pypi"},
    {"npm": "npm"}
  ]
}
```

Reply:
Create a PR for that in intelligence-data

Reply:
I've created an issue, tomorrow I will add the changes
https://github.com/wazuh/intelligence-data/issues/35

Reply:
I've created an issue, tomorrow I will add the changes
https://github.com/wazuh/intelligence-data/issues/35

Reply:
I'll modify the PR according to the new format.